sched/scheduler/jobRunner_test.go

```diff
@@ -40,8 +36,7 @@ func Test_RunTask_PanicWhenWritingStartTaskReturnsFatalError(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
-	// ensures node does not execute task when a fatal error occurs logging starttask
-	nodeMock := cm.NewMockNode(mockCtrl)
+	worker := fake.NewPanicWorker()
```

Comment:
So this works but will get very cumbersome to create a new fake every time you want a slightly different behavior, resulting in  more code to maintain / update.  I prefer mocks, since you get all the behavior at once.  We can discuss on Thursday, this is fine for now